### PR TITLE
PAN-254: Add error handling, improve buttons presentation, add new transaction status value

### PR DESCRIPTION
**Pull request by Meskat**

### CODE REVIEW

File: src/app/advertiser/campaign-details/campaign-details.component.ts [hunk 1]
```diff
@@ -142,15 +143,15 @@ export class CampaignDetailsComponent extends HandleSubscription implements OnIn
...
     } else {
-      this.currentCampaignStatus = 'inactive';
+      status = this.campaignStatusesEnum.INACTIVE;
     }
-    this.campaign.basicInformation.status =
-      this.campaignStatusesEnumArray.findIndex(el => el === this.currentCampaignStatus);
+
     this.store.dispatch(new advertiserActions.UpdateCampaignStatus(
-      {id: this.campaign.id, status: this.campaign.basicInformation.status}));
+      {id: this.campaign.id, status}));
```
Comment (PawelPodkalicki) [inline src/app/advertiser/campaign-details/campaign-details.component.ts]: Is it shortcut for:
```suggestion
      {id: this.campaign.id, status: status}));
```

File: src/app/store/advertiser/advertiser.effects.ts [hunk 2]
```diff
@@ -118,28 +128,28 @@ export class AdvertiserEffects {
...
-            message: `Given campaign cannot be deleted at this moment. Please try again, later`,
-          }
-        });
-      }
-    );
+    .do(payload => {
+      this.dialog.open(ErrorResponseDialogComponent, {
+        data: {
+          title: `Error occurred`,
+          message: `${payload}`,
```
Comment (PawelPodkalicki) [inline src/app/store/advertiser/advertiser.effects.ts]: I am not sure if payload should be displayed as message at this moment. Was that tested? I suggest to replace with generic message like 'Operation could not be executed at this moment. Please try again, later.'.

File: src/app/advertiser/campaign-details/campaign-details.component.ts (hunk 1 above)
Comment (Meskat) [reply src/app/advertiser/campaign-details/campaign-details.component.ts]: yes

File: src/app/store/advertiser/advertiser.effects.ts (hunk 2 above)
Comment (Meskat) [reply src/app/store/advertiser/advertiser.effects.ts]: in this case, payload is a generic message pass as payload to dispatched failure actions

**Pull request merged**